runtime/src/account_storage.rs

```diff
@@ -1,17 +1,73 @@
 //! Manage the map of slot -> append vecs
 
 use {
-    crate::accounts_db::{AccountStorageEntry, AppendVecId, SlotStores},
+    crate::{
+        accounts_db::{AccountStorageEntry, AppendVecId, SlotStores},
+        // TODO(yhchiang): consider moving it to this file.
+        append_vec::{
+            AppendVecAccountsIter, StorableAccountsWithHashesAndWriteVersions, StoredAccountMeta,
+        },
+        storable_accounts::StorableAccounts,
+    },
     dashmap::DashMap,
-    solana_sdk::clock::Slot,
+    solana_sdk::{account::ReadableAccount, clock::Slot, hash::Hash},
     std::{
+        borrow::Borrow,
         collections::HashMap,
+        io::{self},
+        path::{Path, PathBuf},
         sync::{Arc, RwLock},
     },
 };
 
 pub type AccountStorageMap = DashMap<Slot, SlotStores>;
 
+pub trait AccountStorageFile {
```

Comment:
Here I am putting a tentative name.  Any suggestions on the trait name are welcomed.